pkg/plugins/manager/filestore/fs.go

```diff
@@ -0,0 +1,54 @@
+package filestore
+
+import (
+	"context"
+	"io"
+
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/plugins/log"
+	"github.com/grafana/grafana/pkg/plugins/manager/registry"
+)
+
+type Service struct {
+	pluginRegistry registry.Service
+	log            log.Logger
+}
+
+func ProvideService(pluginRegistry registry.Service) *Service {
+	return &Service{
+		pluginRegistry: pluginRegistry,
+		log:            log.New("plugin.fs"),
+	}
+}
+
+func (s *Service) File(ctx context.Context, pluginID, filename string) (*plugins.File, error) {
+	if p, exists := s.pluginRegistry.Plugin(ctx, pluginID); exists {
+		f, err := p.File(filename)
+		if err != nil {
+			return nil, err
+		}
+		defer func() {
+			err = f.Close()
+			if err != nil {
+				s.log.Error("Could not close plugin file", "pluginID", p.ID, "file", filename)
+			}
+		}()
+
+		b, err := io.ReadAll(f)
+		if err != nil {
+			return nil, err
+		}
+
+		fi, err := f.Stat()
+		if err != nil {
+			return nil, err
+		}
+
+		return &plugins.File{
+			Content: b,
```

Comment:
nit: maybe this should be returning the io.Reader and let the user the decission of how to read it? (if files are small enough it's probably not a problem)

Reply:
Good point. I think I'd like to keep it as is now as if files are sent over the wire then we're gonna be dealing in bytes regardless and I'd like the layers to match for simplicity. But I agree it would probably make a nicer API layer eventually!